modules/core/src/main/scala/org/scalasteward/core/nurture/NurtureAlg.scala

```diff
@@ -210,14 +209,35 @@ final class NurtureAlg[F[_]](
     result.flatMap { case (reset, msg) => logger.info(msg).as(reset) }
   }
 
-  def mergeAndApplyAgain(data: UpdateData): F[Unit] =
+  def mergeAndApplyAgain(data: UpdateData): F[ProcessResult] =
     for {
       _ <- logger.info(
         s"Merge branch '${data.baseBranch.name}' into ${data.updateBranch.name} and apply again"
       )
       _ <- gitAlg.mergeTheirs(data.repo, data.baseBranch)
       _ <- editAlg.applyUpdate(data.repo, data.update)
       containsChanges <- gitAlg.containsChanges(data.repo)
-      _ <- if (containsChanges) commitAndPush(data) else F.unit
-    } yield ()
+      result <- if (containsChanges) commitAndPush(data) else F.pure[ProcessResult](Ignored)
+    } yield result
+}
+
+object NurtureAlg {
+  def processUpdates[F[_]](
+      updates: List[Update],
+      updateF: Update => F[ProcessResult],
+      updatesLimit: Option[Int]
+  )(implicit F: Monad[F]): F[Unit] =
+    updates.headOption.fold(F.unit) { up =>
+      updates.drop(1).foldLeft(updateF(up).map(res => updatesLimit.map(_ -> res))) {
+        case (previous, update) =>
+          previous >>= {
+            case None                => updateF(update).map(_ => None)
+            case prev @ Some((0, _)) => F.pure(prev)
+            case Some((remaining, Ignored)) =>
+              updateF(update).map(result => (remaining -> result).some)
+            case Some((remaining, Updated)) =>
+              updateF(update).map(result => (remaining - 1 -> result).some)
+          }
+      } >> F.unit
+    }
```

Comment:
I imagine this could be easier implemented with a `fs2.Stream`. Here is my idea:
 - `Stream.emits(updates)`
 - `evalMap` the stream with `updateF`
 - map `Updated` to 1 and `Ignored` to 0
 - transform the stream to the running sum of the 1s and 0s (with `scan(0)(_ + _)` or `scanMonoid`)
 - `takeWhile` the running sum is smaller than `updatesLimit`

What do you think?

Reply:
Sounds good.